### ISSUE TITLE
What does the "This benchmark provides no result that can be validated" warning mean?

### ISSUE THREAD

**Issue by tactlessfish** [renaissance-benchmarks/renaissance#334]
I've tried a few benchmarks, and they all output
```
WARNING: This benchmark provides no result that can be validated.
There is no way to check that no silent failure occurred.
```
If I'm understanding this correctly, a benchmark could fail partway through and have a lower duration than a benchmark that ran successfully.

What does this mean for me? Do I have to disregard the benchmark results if they can't be validated?

**Comment by ceresek** [comment]
Ideally, each repetition of a benchmark would consist of a "work phase", where the measured work is done, and a "validation phase", where the result of the work phase is checked for correctness, simply to make sure the work that was supposed to be done in the work phase was really executed. Unfortunately, for some of our benchmarks it is not easy to validate the results (especially in a reliable platform portable way), and for some others we simply did not yet get to implementing the validation. Those benchmarks now print the warning you refer to.

What does this mean for your benchmark results ? Most likely, nothing bad. We did check all the benchmarks in typical conditions and although we may not have the validation implemented, we are reasonably confident that the benchmarks execute the intended work phases. However, there may be situations that may require attention - for example, if you execute the benchmark in an unusual configuration or on an unusual platform (we test most with OpenJDK and HotSpotJDK on Linux), if you see the benchmark throw exceptions during execution, if you use a development (rather than stable) version of your JDK or other involved software - these are all situations that might reveal some corner case in the benchmark or the harness that we did not cover, and thus invalidate the results.

To sum up - the warning simply says that if you need to be really really sure that the benchmarks worked as intended, you should make extra steps to check everything worked - obviously, you will need to consider the effort invested vs the level of certainty you need.

P.S.: Did I mention we accept pull requests for implementations of the missing validation phases ? :-)

**Comment by tactlessfish** [comment]
Thanks, that clears it up nicely!
